### PR TITLE
Updated Key-Value extension to use a list view.

### PR DESCRIPTION
**Pull request by gkpln3**
## Description 

Improved the key-value extension to use a list view.
Approved by extension author @NirLevy98.

## Screencast

https://github.com/raycast/extensions/assets/8081679/f0cb86f4-e1cc-449c-a4c0-db8a90bf5fe5

## Checklist

- [X] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [X] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [X] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [X] I checked that files in the `assets` folder are used by the extension itself
- [X] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by NirLevy98** [comment]
A very useful addition!

**Comment by pernielsentikaer** [comment]
That's indeed a nice update, I think it could be simplified a bit with this approach, what do you two thinks?

```
import { ActionPanel, List, Action, getPreferenceValues } from "@raycast/api";
import { useCachedPromise } from "@raycast/utils";
import * as fs from "fs";

export default function Command() {
  const prefs = getPreferenceValues<ExtensionPreferences>();

  const { isLoading, data } = useCachedPromise(
    async () => {
      return JSON.parse(fs.readFileSync(prefs.jsonFilePath, "utf8"));
    },
    [],
    { initialData: [] }
  );

  const createListItem = (key: string, value: string) => {
    return (
      <List.Item
        key={key}
        title={key}
        keywords={[value]}
        subtitle={value}
        actions={
          <ActionPanel>
            <Action.CopyToClipboard content={value} />
          </ActionPanel>
        }
      />
    );
  };

  return (
    <List isLoading={isLoading}>
      <List.Section title="Key -> Value">{Object.keys(data).map((key) => createListItem(key, data[key]))}</List.Section>
    </List>
  );
}
```

**Comment by gkpln3** [comment]
Definitely, React isn't my expertise 😅
I'll commit an update

**Comment by pernielsentikaer** [comment]
There might be one thing I forgot about, it's the copy to clipboard - two solutions:

* Roll back and have to lists
* Have two actions on the list item (One for copy key, one for copy value)

**Comment by gkpln3** [comment]
@pernielsentikaer I am adding two sections, one for key -> value and one for value -> key
The goal is to be able to quickly convert between the two quickly, so if i paste one i immediately get the other.

File: extensions/key-value/src/index.tsx [hunk 1]
```diff
@@ -1,49 +1,38 @@
...
+    [],
+    { initialData: [] }
+  );
+
+  const createListItem = (key: string, value: string) => {
+    return (
+      <List.Item
+        key={key}
+        title={key}
+        keywords={[value]}
```
[the remark this answers is not in the document]
Comment (gkpln3) [reply extensions/key-value/src/index.tsx:21]: Nope, forgot to remove it 😅
fixing...

**Review by pernielsentikaer — APPROVED** [review]
Hi 👋 

Looks good to me, approved 🔥

**Pull request merged**